pkg/testcase/upgradenodereplacement.go

```diff
@@ -598,3 +598,25 @@ func buildRke2Cmd(
 
 	return cmd, nil
 }
+
+// DeleteReplacedNodes deletes the new nodes created as replacements.
+func DeleteReplacedNodes(cluster *shared.Cluster) {
```

Comment:
this really should not live here as a test, you are narrow it down too much

this should live on cluster config, and we have destroy cluster there, we should have a function `deleteAwsInstances` and call it wherever we want on suites that might need ,  not being specific for node replace

Reply:
So this cannot be added to shared/clusterconfig.go due to circular dependency with aws.go... any other suggestion where this can live?

Reply:
yeah makes sense. since this is actual aws implementation, should live in aws.go? 

then on suite 
```

var _ = AfterSuite(func() {
	if customflag.ServiceFlag.Destroy {
		aws.DeleteAwsNodes(cluster)
	}
})

```

Reply:
also on suite you will need to a var for aws here 
```
var (
	kubeconfig string
	flags      *customflag.FlagConfig
	cluster    *shared.Cluster
)
```

to be able to reuse it on delete function and also pass it to `testcase.TestUpgradeReplaceNode(cluster,aws,  flags)`

Reply:
So we already have `DeleteInstance` in aws.go, which I have used in this func so I don't think it should live there as it deals with cluster stuff... one of the reason I wanted to have a `helper`/`support` dir that will live along with or close to the testcases

Reply:
upgradeerplacement_test.go
```


.....



var _ = AfterAll(func() {
	ips := append(cluster.ServerIPs, cluster.AgentIPs...)
	for _, ip := range ips {
		err := aws.DeleteInstance(ip)
		Expect(err).ToNot(HaveOccurred())
	}
})


```

Reply:
suite
```
var (
	kubeconfig string
	flags      *customflag.FlagConfig
	cluster    *shared.Cluster
	aws        *c.Client
)

func TestMain(m *testing.M) {
	flags = &customflag.ServiceFlag
	flag.Var(&flags.InstallMode, "installVersionOrCommit", "Upgrade with version or commit")
	flag.Var(&flags.Channel, "channel", "channel to use on upgrade")
	flag.Var(&flags.Destroy, "destroy", "Destroy cluster after test")
	flag.Var(&flags.SUCUpgradeVersion, "sucUpgradeVersion", "Version for upgrading using SUC")
	flag.Parse()

	_, err := config.AddEnv()
	if err != nil {
		shared.LogLevel("error", "error adding env vars: %w\n", err)
		os.Exit(1)
	}

	kubeconfig = os.Getenv("KUBE_CONFIG")
	if kubeconfig == "" {
		// gets a cluster from terraform.
		cluster = shared.ClusterConfig()
	} else {
		// gets a cluster from kubeconfig.
		cluster = shared.KubeConfigCluster(kubeconfig)
	}

	aws, err = c.AddClient(cluster)
	Expect(err).NotTo(HaveOccurred())

	os.Exit(m.Run())
}
```